File: Đề NĐ các năm/20-21/Bai XTB 2021/I91202/GHH.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main()
{
    int n;
    int a[10000];
    ifstream f("ghh.inp");
    f>>n;
    for (int i=0;i<n;i++)
        f>>a[i];
    ofstream g("ghh.out");
    g<<2<<"\n";g<<12<<"\n";g<<6<<"\n";
    f.close();
    g.close();
    return 0;
}

File: Đề NĐ các năm/20-21/Bai XTB 2021/I64203/GHH.pas
program b1;
var fi,fo:text;
    i,n,d,x:longint;
    a:array[1..10000] of longint;
    function t(x:longint):int64;
    var i:longint;
    begin
     t:=0;
     for i:=1 to trunc(sqrt(x)) do
      if x mod i=0 then t:=t+i+x div i;
     if sqr(trunc(sqrt(x)))=x then t:=t-trunc(sqrt(x));
    end;
begin
 assign(fi,'GHH.inp'); assign(fo,'GHH.out');
 reset(fi); rewrite(fo); readln(fi,n); d:=0;
 for i:=1 to n do
  begin
   readln(fi,x);
   if 2*x<=t(x) then begin d:=d+1; a[d]:=x; end;
  end;
 writeln(fo,d);
 for i:=1 to d do writeln(fo,a[i]);
 close(fi); close(fo);
end.
